src/Features/Core/Portable/EmbeddedLanguages/Json/JsonParser.JsonNetSyntaxChecks.cs

```diff
@@ -0,0 +1,236 @@
+                    CultureInfo.InvariantCulture, out _))
+                {
+                    return new EmbeddedDiagnostic(
+                        FeaturesResources.Invalid_number,
+                        GetSpan(chars));
+                }
+
+                return null;
+            }
+
+            private static EmbeddedDiagnostic? CheckArray(JsonArrayNode node)
+            {
+                foreach (var child in node.Sequence)
+                {
+                    if (child.Kind == JsonKind.Property)
+                    {
+                        return new EmbeddedDiagnostic(
+                            FeaturesResources.Properties_not_allowed_in_an_array,
+                            ((JsonPropertyNode)child).ColonToken.GetSpan());
+                    }
+                }
+
+                var diagnostic = CheckCommasBetweenSequenceElements(node.Sequence);
+                return diagnostic ?? CheckChildren(node);
+            }
+
+            private static EmbeddedDiagnostic? CheckConstructor(JsonConstructorNode node)
+            {
+                if (!IsValidConstructorName(node.NameToken))
+                {
+                    return new EmbeddedDiagnostic(
+                        FeaturesResources.Invalid_constructor_name,
+                        node.NameToken.GetSpan());
+                }
+
+                return CheckCommasBetweenSequenceElements(node.Sequence) ?? CheckChildren(node);
+            }
+
+            private static bool IsValidConstructorName(JsonToken nameToken)
+            {
+                foreach (var vc in nameToken.VirtualChars)
+                {
+                    if (!vc.IsLetterOrDigit)
+                        return false;
+                }
+
+                return true;
+            }
+
+            private static EmbeddedDiagnostic? CheckCommasBetweenSequenceElements(ImmutableArray<JsonValueNode> sequence)
+            {
+                // Json.net allows sequences of commas.  But after every non-comma value, you need
+                // a comma.
+                for (int i = 0, n = sequence.Length - 1; i < n; i++)
+                {
+                    var child = sequence[i];
+                    var nextChild = sequence[i + 1];
+                    if (child.Kind != JsonKind.CommaValue &&
+                        nextChild.Kind != JsonKind.CommaValue)
+                    {
+                        return new EmbeddedDiagnostic(
+                           string.Format(FeaturesResources._0_expected, ','),
+                           GetFirstToken(nextChild).GetSpan());
+                    }
+                }
+
+                return null;
+            }
+
+            private static EmbeddedDiagnostic? CheckObject(JsonObjectNode node)
+            {
+                for (int i = 0, n = node.Sequence.Length; i < n; i++)
+                {
+                    var child = node.Sequence[i];
+                    if (i % 2 == 0)
```

Comment:
Do objects not allow for multiple comma entries like `CheckCommasBetweenSequenceElements`? Can this be moved to a `EmbeddedSeparatedSyntaxNodeList`?

Reply:
json.net allows for commas in a row :)

Reply:
Does this though? :) It looks like all even items must be `JsonKind.Property`

Reply:
ah.  it accepts that for arrays... not objects.  i'll see how feasible it is to try for a separated list here!